java-client/src/main/java/com/cumulocity/sdk/client/notification/SubscriberImpl.java

```diff
@@ -265,13 +280,18 @@ public boolean rcvMeta(ClientSession session, Mutable message) {
             } else {
                 return true;
             }
-            // Resubscribe only there is a successful handshake and successfully connected.
+            // Resubscribe all only there is a successful handshake and successfully connected.
             if (reHandshakeSuccessful && reconnectedSuccessful) {
                 log.debug("reconnect operation detected for session {} - {} ", bayeuxSessionProvider, session.getId());
                 reHandshakeSuccessful = false;
                 reconnectedSuccessful = false;
                 resubscribeAll();
             }
+            // Subscribe pending on each successful /meta/connect
+            if (reconnectedSuccessful) {
+                reconnectedSuccessful = false;
+                resubscribePending();
```

Comment:
so this is the case when we are having connected but we are not subscribed?

Reply:
Yeah, on successful `/meta/connect` we should process all `pendingSubscriptions`, before this fix we've waited the whole `/meta/handshake` && `/meta/connect` cycle. But the way this code works we often could first do `/meta/handshake` and `/meta/connect` (first `/meta/connect` with advice) and only then send `/meta/subscribe` (or receive unsuccessul response to it)